analyzer.go

```diff
@@ -15,41 +16,182 @@ type Platform interface {
 	API() *api.Version
 }
 
+type AnalyzerFactory struct {
+	platformAPI     *api.Version
+	cacheHandler    CacheHandler
+	configHandler   ConfigHandler
+	imageHandler    ImageHandler
+	registryHandler RegistryHandler
+}
+
+func NewAnalyzerFactory(platformAPI *api.Version, cacheHandler CacheHandler, configHandler ConfigHandler, imageHandler ImageHandler, registryHandler RegistryHandler) *AnalyzerFactory {
+	return &AnalyzerFactory{
+		platformAPI:     platformAPI,
+		cacheHandler:    cacheHandler,
+		configHandler:   configHandler,
+		imageHandler:    imageHandler,
+		registryHandler: registryHandler,
+	}
+}
+
 type Analyzer struct {
 	PreviousImage imgutil.Image
 	RunImage      imgutil.Image
 	Logger        Logger
-	Platform      Platform
 	SBOMRestorer  layer.SBOMRestorer
 
 	// Platform API < 0.7
 	Buildpacks            []buildpack.GroupBuildpack
 	Cache                 Cache
 	LayerMetadataRestorer layer.MetadataRestorer
+	RestoresLayerMetadata bool
+}
+
+func (f *AnalyzerFactory) NewAnalyzer(
```

Comment:
In the past, adding new inputs to the analyzer/restorer/etc. was a source of bugs because we would forget to update the creator. Accepting the inputs as a list, while ugly, can avoid this problem.